### PR TITLE
Mule 9664

### PR DESCRIPTION
**Pull request by pablomusumeci**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java [hunk 1]
```diff
@@ -336,7 +337,11 @@ private void declareMessageSource(HasSourceDeclarer declarer, Class<? extends So
                 .withMetadataResolverFactory(getMetadataResolverFactoryFromClass(extensionType, sourceType));
 
         declareMetadataKeyId(sourceType, source);
-        declareSingleParameters(getParameterFields(sourceType), source);
+        declareSingleParameters(getParameterFields(sourceType), source,
+                                MuleExtensionAnnotationParser::parseMetadataAnnotations);
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java]: inline

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/MuleExtensionAnnotationParser.java [hunk 2]
```diff
@@ -327,4 +330,14 @@ private static boolean isParameterFieldContainer(Set<Class<? extends Annotation>
     {
         return (annotations.contains(ParameterGroup.class) || annotations.contains(MetadataKeyId.class)) && parameterType instanceof ObjectType;
     }
+
+    public static void parseConnectionAnnotation(Class<?> annotatedFieldClass, HasModelProperties parameter)
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/MuleExtensionAnnotationParser.java]: can't thse two methods have package visibility?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java [hunk 3]
```diff
@@ -350,6 +351,7 @@ private static boolean isVoid(Class<?> type)
         return getAnnotatedFields(extensionType, Parameter.class);
     }
 
+
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java]: remove empty line

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MuleExtensionUtils.java [hunk 4]
```diff
@@ -126,10 +129,35 @@ public static boolean acceptsExpressions(ExpressionSupport support)
...
+    public static List<SourceModel> getConnectedSources(RuntimeConfigurationModel configurationModel)
+    {
+        List<SourceModel> sources = new LinkedList<>();
+        sources.addAll(collectConnectedSources(configurationModel.getExtensionModel()));
+        sources.addAll(collectConnectedSources(configurationModel));
+
+        return sources;
+    }
+
+    public static List<ComponentModel> getConnectedComponents(RuntimeConfigurationModel configurationModel)
```
Comment (estebanwasinger) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/MuleExtensionUtils.java]: Javadoc

File: modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/config/SourceWithoutConnectedOperationTestCase.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import org.junit.Test;
+
+public class SourceWithoutConnectedOperationTestCase extends ExtensionFunctionalTestCase
+{
+    @Override
+    protected String getConfigFile()
+    {
+        return "vegan-config.xml";
+    }
+
```
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/java/org/mule/runtime/module/extension/internal/config/SourceWithoutConnectedOperationTestCase.java]: you need to override the `doSetUpBeforeMuleContextCreation` method and set `HarvestPeachesSource.isConnected`  to `false`

File: modules/extensions-spring-support/src/test/resources/vegan-config.xml [hunk 6]
```diff
@@ -22,6 +26,12 @@
         <vegan:eat-kiwi config-ref="kiwi"/>
     </flow>
 
+    <flow name="peachOk">
+        <vegan:harvest-peaches config-ref="peach"/>
+        <!-- intentionally left without operations-->
```
Comment (estebanwasinger) [inline modules/extensions-spring-support/src/test/resources/vegan-config.xml]: I don't think to be necessary this comment

**Pull request merged**